### ISSUE TITLE
Middy does not export types correctly

### ISSUE THREAD

**Issue by acommodari** [middyjs/middy#1071]
**Describe the bug**

If a project is using `commonjs` as well as `module`/`moduleResolution` `node16` or `nodenext` in their `tsconfig.json` we get the following error:
![image](https://github.com/middyjs/middy/assets/14224801/d34e8261-5b4d-43d3-9c61-7f04e65c8af6)

**To Reproduce**
How to reproduce the behaviour:
1. create a dummy node project in `commonjs` with TypeScript where `moduleResolution` is set to `node16` or `nodenext`.
2. Try importing `@middy/core`

**Expected behaviour**

Types should work properly if TS `module` and `moduleResolution` are set to `node16` or `nodenext`.

**Environment (please complete the following information):**
 - Node.js: 18.16.1
 - Middy: 4.5.2
 - AWS SDK 3.x

**Additional context**
You can test packages using this site which scans your package.json and it also reports that this issue will occur:
https://arethetypeswrong.github.io/?p=%40middy%2Fcore%404.5.2

You can also follow this amazing post of Twitter by a TypeScript maintainer that explains this issue: 
https://twitter.com/atcb/status/1634653474041503744

The TLDR is that you need to generate 2 separate sets of types one for ESM and one for CJS and right now you only have ESM which is whats causing the issue. He also recommends doing away with the types fields all together. But yes the main solution would be to generate a second set of types.

**Comment by willfarrell** [comment]
Thanks for reporting this. The inclusion of the `types` field came from the TypeScript documentation ... This recommended change might have unexpected consequences for others and thus would be a breaking change. Currently I'm planning to deprecate cjs support in the next version, planned for this fall, making the fix for this unnecessary. Unfortunately, I have to put a won't fix on this issue for now, sorry. If we get a large sponsor that says cjs is really important to them, we could delay it's deprecation and revisit this issue.

Anyway, I've made a note in our TS rewire issue to review this as part of that work and ensure we're following the latest best practice.

**Comment by acommodari** [comment]
Thank you for the extremely quick reply! While it's unfortunate this won't be fixed I also understand the motivation to go ESM only. Hopefully we can migrate our project to ESM in the future so that we can make use of your great tool!
